plugins/test/framework.cc

```diff
@@ -55,6 +96,26 @@ ContextOptions& TestContext::options() const {
   return static_cast<TestWasm*>(wasm())->options();
 }
 
+proxy_wasm::BufferInterface* TestHttpContext::getBuffer(
+    proxy_wasm::WasmBufferType type) {
+  switch (type) {
+    case proxy_wasm::WasmBufferType::PluginConfiguration:
+      return TestContext::getBuffer(type);
+    case proxy_wasm::WasmBufferType::HttpRequestBody:
+      if (current_callback_ == TestHttpContext::CallbackType::RequestBody) {
+        return &body_buffer_;
+      }
+      break;
+    case proxy_wasm::WasmBufferType::HttpResponseBody:
+      if (current_callback_ == TestHttpContext::CallbackType::ResponseBody) {
+        return &body_buffer_;
+      }
+      break;
+    default:
+      return nullptr;
+  }
+  return nullptr;
```

Comment:
nit: since the switch statement is exhaustive, this line can't be reached, so you should be able to remove the final `return nullptr;` that is outside of the switch statement.

Reply:
AIUI if you were to break out of that switch statement then the default case would not be called, so we would need to return outside of the switch.